ports/stm32/mpconfigport.h

```diff
@@ -342,6 +342,9 @@ struct _mp_bluetooth_btstack_root_pointers_t;
     /* pointers to all CAN objects (if they have been created) */ \
     struct _pyb_can_obj_t *pyb_can_obj_all[MICROPY_HW_MAX_CAN]; \
     \
+    /* pointers to all I2S objects (if they have been created) */ \
+    struct _machine_i2s_obj_t *machine_i2s_obj[MICROPY_HW_MAX_I2S]; \
```

Comment:
I'm not sure 0-length arrays are allowed, but `MICROPY_HW_MAX_CAN` can be 0 so it must be ok?

Reply:
Based on the CAN implementation (which I used as a pattern to implement I2S root pointers), it seems like 0 length arrays are allowed.    It looks suspect, but this approach provides an elegant way to compile ports that don't support a given peripheral, such as CAN and I2S.  I tried compiling several STM32 ports that lack I2S support and I never saw a compile error.

Reply:
Ok, let's leave it then.  Can improve later if needed.